fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcMySQLClient.java

```diff
@@ -294,30 +296,33 @@ private boolean isConvertDatetimeToNull(JdbcClientConfig jdbcClientConfig) {
      * get all columns like DatabaseMetaData.getColumns in mysql-jdbc-connector
      */
     private Map<String, String> getColumnsDataTypeUseQuery(String remoteDbName, String remoteTableName) {
-        Connection conn = getConnection();
+        Connection conn = null;
+        Statement stmt = null;
         ResultSet resultSet = null;
-        Map<String, String> fieldtoType = Maps.newHashMap();
+        Map<String, String> fieldToType = Maps.newHashMap();
 
         StringBuilder queryBuf = new StringBuilder("SHOW FULL COLUMNS FROM ");
         queryBuf.append(remoteTableName);
         queryBuf.append(" FROM ");
         queryBuf.append(remoteDbName);
-        try (Statement stmt = conn.createStatement()) {
+        try {
+            conn = getConnection();
+            stmt = conn.createStatement();
             resultSet = stmt.executeQuery(queryBuf.toString());
             while (resultSet.next()) {
                 // get column name
                 String fieldName = resultSet.getString("Field");
                 // get original type name
                 String typeName = resultSet.getString("Type");
-                fieldtoType.put(fieldName, typeName);
+                fieldToType.put(fieldName, typeName);
             }
         } catch (SQLException e) {
             throw new JdbcClientException("failed to get jdbc columns info for remote table `%s.%s`: %s",
                     remoteDbName, remoteTableName, Util.getRootCauseMessage(e));
         } finally {
-            close(resultSet, conn);
+            close(resultSet, stmt, conn);
         }
-        return fieldtoType;
+        return fieldToType;
```

Comment:
Using try-with-resources is preferable; otherwise, you have to be extra careful about potential null pointers. In a finally block, you’d need to check if each item is null before closing it,

```
finally {
    // Null check before closing each resource
    if (resultSet != null) {
        try {
            resultSet.close();
        } catch (SQLException e) {
            LOG.warn("Failed to close ResultSet", e);
        }
    }
    if (stmt != null) {
        try {
            stmt.close();
        } catch (SQLException e) {
            LOG.warn("Failed to close Statement", e);
        }
    }
    if (conn != null) {
        try {
            conn.close();
        } catch (SQLException e) {
            LOG.warn("Failed to close Connection", e);
        }
    }
}
```

 which makes the code messy.

```
try (Connection conn = getConnection();
     Statement stmt = conn.createStatement();
     ResultSet resultSet = stmt.executeQuery(queryBuf.toString())) {

    while (resultSet.next()) {

        String fieldName = resultSet.getString("Field");
        String typeName = resultSet.getString("Type");
        fieldtoType.put(fieldName, typeName);
    }
} catch (SQLException e) {
    throw new JdbcClientException(String.format("failed to get jdbc columns info for remote table `%s.%s`: %s",
            remoteDbName, remoteTableName, Util.getRootCauseMessage(e)), e);
}
```

Reply:
I checked whether the object is null in the close() method, and because some drivers do not implement AutoCloseable, we need to close it manually.